packages/xrpl/src/Wallet/index.ts

```diff
@@ -197,6 +202,35 @@ class Wallet {
     })
   }
 
+  /**
+   * Derives a wallet from secret numbers.
+   *
+   * @param secretNumbers - A string consisting of 8 times 6 numbers (whitespace delimited) used to derive a wallet.
+   * @param opts - (Optional) Options to derive a Wallet.
+   * @param opts.masterAddress - Include if a Wallet uses a Regular Key Pair. It must be the master address of the account.
+   * @returns A Wallet derived from secret numbers.
+   * @throws ValidationError if unable to derive private key from secret number input.
+   */
+   public static fromSecretNumbers(
+    secretNumbers: Array<string> | string,
+    opts: { masterAddress?: string, algorithm?: ECDSA } = {},
+  ): Wallet {
+    let numbersArray:Array<string> = [];
+
+    if (typeof secretNumbers === 'string') {
+      numbersArray = Utils.parseSecretString(secretNumbers);
+    } else if (Array.isArray(secretNumbers)) {
+      numbersArray = secretNumbers;
+    }
```

Comment:
Since some users might be using javascript and pass in something unexpected, it's probably worth adding an exception if the type doesn't match. 
```suggestion
    } else {
         throw new ValidationError("Unable to parse the given secret numbers.")
    }
```